nailgun/entities.py

```diff
@@ -6912,3 +6912,112 @@ def update(self, fields=None):
         """
         self.update_json(fields)
         return self.read()
+
+
+class VirtWhoConfig(
+        Entity,
+        EntityCreateMixin,
+        EntityDeleteMixin,
+        EntityReadMixin,
+        EntitySearchMixin,
+        EntityUpdateMixin):
+    """A representation of a VirtWho Config entity."""
+
+    def __init__(self, server_config=None, **kwargs):
+        self._fields = {
+            'blacklist': entity_fields.StringField(),
+            'debug': entity_fields.BooleanField(),
+            'filtering_mode': entity_fields.IntegerField(
+                choices=[0, 1, 2], default=0, required=True),
+            'hypervisor_id': entity_fields.StringField(
+                choices=['hostname', 'uuid', 'hwuuid'],
+                default='hostname', required=True),
+            'hypervisor_password': entity_fields.StringField(),
+            'hypervisor_server': entity_fields.StringField(required=True),
+            'hypervisor_type': entity_fields.StringField(
+                choices=['esx', 'rhevm', 'hyperv', 'xen', 'libvirt'],
+                default='libvirt', required=True),
+            'hypervisor_username': entity_fields.StringField(required=True),
+            'interval': entity_fields.IntegerField(
+                choices=[60, 120, 240, 480, 720], default=120, required=True),
+            'name': entity_fields.StringField(required=True),
+            'no_proxy': entity_fields.StringField(),
+            'organization_id': entity_fields.IntegerField(required=True),
+            'proxy': entity_fields.StringField(),
+            'satellite_url': entity_fields.StringField(required=True),
+            'whitelist': entity_fields.StringField(),
+            'organization': entity_fields.OneToOneField(Organization)
+        }
+        self._meta = {
+            'api_path': 'foreman_virt_who_configure/api/v2/configs',
+            'server_modes': ('sat', 'sam'),
+        }
+        super(VirtWhoConfig, self).__init__(server_config, **kwargs)
+
+    def path(self, which=None):
+        """Extend ``nailgun.entity_mixins.Entity.path``.
+
+        The format of the returned path depends on the value of ``which``:
+
+        deploy_script
+            /foreman_virt_who_configure/api/v2/configs/:id/deploy_script
+
+        ``super`` is called otherwise.
+
+        """
+        if which in ('deploy_script'):
+            return '{0}/{1}'.format(
+                super(VirtWhoConfig, self).path(which='self'), which)
+        return super(VirtWhoConfig, self).path(which)
+
+    def create_payload(self):
+        """
+        Wraps config in extra dict
+        :return:
+        """
+        return {'foreman_virt_who_configure_config':
+                super(VirtWhoConfig, self).create_payload()}
+
+    def deploy_script(self, synchronous=True, **kwargs):
```

Comment:
Is `synchronous` required? I don't think this API is async

Reply:
This is an artifact of using APIx to generate the entity. APIx's initial development for nailgun's template included asynchronous methods.

Reply:
Removed the async stuff